node/src/actors/chain_manager/mod.rs

```diff
@@ -811,6 +811,104 @@ impl ChainManager {
         self.magic
     }
 
+    /// Construct superblock process which uses futures
+    pub fn construct_superblock(
+        &mut self,
+        ctx: &mut Context<Self>,
+        block_epoch: u32,
+    ) -> ResponseActFuture<Self, Option<SuperBlock>, ()> {
+        let consensus_constants = self
+            .chain_state
+            .chain_info
+            .as_ref()
+            .unwrap()
+            .consensus_constants
+            .clone();
```

Comment:
There is a method that do the same
```suggestion
        let consensus_constants = self.consensus_constants();
```